src/async-input.js

```diff
@@ -15,31 +15,48 @@
  */
 
 /**
- * Base Class for all AMP Async Input Elements.
- * This simply Extends the Base Element, and adds a new
- * overridable function, getValue(). This should be
- * used by components like AMP form, to async request
+ * Interface for all AMP Async Input Elements.
+ * enforces the overridable function, getValue().
+ * Async Input should be implemented
+ * by components like AMP form, to async request
  * a value from a component, and then be used for
  * some other action. For examples, this can be used
  * by reCAPTCHA to request tokens for the form.
+ *
+ * NOTE: Elements that implemented AsyncInput must
+ * Also add and follow the other exported constants.
+ * See amp-recaptcha-input as an example.
+ *
+ * @interface
  */
-import {BaseElement} from './base-element';
-
-export class AsyncInput extends BaseElement {
-
-  /** @param {!AmpElement} element */
-  constructor(element) {
-    super(element);
-    element.classList.add('i-amphtml-async-input');
-  }
+export class AsyncInput {
 
   /**
    * Called to get the asynchronous value of an
    * AsyncInput field.
    * @return {!Promise<string>}
    */
-  getValue() {
-    throw new Error('Elements that implement AsyncInput' +
-      ' must override getValue()');
-  }
+  getValue() {}
 }
+
+/**
+ * Classes
+ *
+ * Components implementing the AsyncInput,
+ * are expected to add the following classes
+ * at the appropriate time.
+ *
+ * @constant {!Object<string, string>}
+ */
+export const AsyncInputClasses = {
+  /**
+   * i-amphtml-async-input
+   *
+   * Required base class that must be added to the async-input
+   * element on buildCallback or layoutCallback.
+   * This will be used by other amp components to find
+   * and use async-input elements.
+   */
+  ASYNC_INPUT: 'i-amphtml-async-input',
```

Comment:
Do we have to preserve keys across boundaries? If so we will need to use `'ASYNC_INPUT'`